crates/lib/src/qpu/api.rs

```diff
@@ -404,6 +421,17 @@ impl ApiExecutionOptionsBuilder {
             .bypass_settings_protection = bypass_settings_protection;
         self
     }
+
+    /// Set the `timeout` value. See [`ApiExecutionOptions::timeout`] for more information.
+    pub fn timeout(
+        &mut self,
+        timeout: Option<::pbjson_types::Duration>,
+    ) -> Result<&mut Self, QpuApiError> {
```

Comment:
Looks like this is infallible; why return a `Result`?

Reply:
Leftover from a previous implementation where I had `std::time::Duration` in the signature, which was a fallible conversion. I reverted that since pyo3 doesn't support Duration.